### PR TITLE
Add realtime sequencing ability

### PR DESCRIPTION
**Pull request by po09i**
## Checklist

- [x] I've linked relevant issues in the PR body
- [x] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

## Description
We currently have a sequencer to perform slicewise shimming. This PR introduces the following:

* Better selection of slices
  * Multi-slice: [(0, 3), (1, 4), (2, 5)] → slices 0 and 3 are optimized at the same time
  * Slab: [(0, 1, 2, 3, 4, 5, 6)] → slab with all slices
  * Slicewise: [(0,), (1,), (2,), (3,), (4,), (5,), (6,)] → slice by slice
  * API to select slices: define_slices(...)
* Refactoring of the sequencer files (shim_sequencer())
* Realtime API to perform realtime shimming (shim_realtime_pmu_sequencer())
* Dilation of masks
* Added min/max attributes to the pmu

## How to test
Debug `test_sequencer.test_realtime_sequencer()`

### CODE REVIEW

File: shimmingtoolbox/shim/sequencer.py [hunk 1]
```diff
@@ -64,9 +64,10 @@ def shim_realtime_pmu_sequencer(nii_fieldmap, json_fmap, pmu: PmuResp, coils: Li
     Returns:
         (tuple): tuple containing:
 
-            * numpy.ndarray: Static coefficients to shim (len(slices) x channels)
-            * numpy.ndarray: Static coefficients to shim (len(slices) x channels)
-
+            * numpy.ndarray: Static coefficients to shim (len(slices) x channels) [hz]
+            * numpy.ndarray: Static coefficients to shim (len(slices) x channels) [hz/unit_pressure]
```
Comment (gaspardcereza) [inline shimmingtoolbox/shim/sequencer.py]: Minor detail: The symbol for Hertz is Hz not hz

**Comment by po09i** [comment]
I implemented a realtime shimming sequencer (`shim_realtime_pmu_sequencer`). I showed a couple of you preliminary results, you can look at [slide 8 and 9](https://docs.google.com/presentation/d/1nC9XmDUW47iq1qYfrkpUpjL2o3paB1c1NHxORTuePIY/edit?usp=sharing) for updated results. I'd also appreciate feedback on the code. Once I have feedback I will clean up a bit more the test file.

My plan is to then implement a shim API which will combine both realtime and "one timepoint" API so we can call this in a CLI.

File: shimmingtoolbox/optimizer/basic_optimizer.py [hunk 2]
```diff
@@ -109,15 +132,48 @@ def merge_coils(self, unshimmed, affine):
...
             for a_bound in coil.coef_channel_minmax:
                 bounds.append(a_bound)
 
-        coil_profiles = np.concatenate(coil_profiles_list, axis=3)
+        return bounds
 
-        return coil_profiles, bounds
+    def initial_guess_half_bounds(self):
+        """
+        Calculates the initial guess from the bounds, sets it to half of the bounds
```
Comment (gaspardcereza) [inline shimmingtoolbox/optimizer/basic_optimizer.py]: It does not seem to set it to half the bounds but to the average between the bounds

**Review by gaspardcereza — APPROVED** [review]
LGTM (just added a minor comment )

**Review by jcohenadad — APPROVED** [review]

File: shimmingtoolbox/masking/mask_utils.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+                      np.array([[[0, 0, 0],
+                                 [0, 1, 0],
+                                 [0, 0, 0]],
+                                [[0, 1, 0],
+                                 [1, 1, 1],
+                                 [0, 1, 0]],
+                                [[0, 0, 0],
+                                 [0, 1, 0],
+                                 [0, 0, 0]]])
```
Comment (gaspardcereza) [inline shimmingtoolbox/masking/mask_utils.py:88]: Is `cross[2]` the same as `cross[0]` or is it a mistake ? Also the size of the kernel is only specified for `sphere`.

Comment (po09i) [reply shimmingtoolbox/masking/mask_utils.py:88]: Its a 3x3 kernel where cross[0] == cross[2]. I added a size comment for the other ones.

**Pull request merged**